### ISSUE TITLE
Code action 'Create method...' does not work

### ISSUE THREAD

**Issue by loralei28** [redhat-developer/vscode-java#3780]
##### Environment
- Operating System: Windows
- JDK version:17
- Visual Studio Code version:1.93.1
- Java extension version: redhat-1.35.0

##### Steps To Reproduce
1. Create a new Java file, for example Test.java
2. In Test.java file, invoke `class` and `main` code snippet to generate code
3. Type `bar();`
4. Click ‘Quick fix..’ in the java file or click 'Show fix' in the error window
5. Select code action "Create method 'bar()'"

##### Current Result

”Create method 'bar()'“ does not work and will delete "bar()"
![3](https://github.com/user-attachments/assets/9a374ba3-67eb-4873-8995-cd2727d2676e)

**Comment by rgrunber** [comment]
https://github.com/user-attachments/assets/dda5bdb4-67c2-4b34-b9aa-fcad49d4a91e

Seems to be working for me. @loralei28 , @testforstephen what am I missing here. Also, if you activate `Java: Open Java Language Server Log File`, are there any errors in that log ? Does the "Extension Host" logs (drom the dropdown) of the Output view indicate any errors ?

**Comment by datho7561** [comment]
Here is a sample file:

```java
package functions;

import io.quarkus.funqy.Funq;

/**
 * Your Function class
 */
public class Function {

    /**
     * Use the Quarkus Funqy extension for our function. This function simply echoes its input
     * @param input a Java bean
     * @return a Java bean
     */
    @Funq
    public Output function(Input input) {

        // Add business logic here
        bar();

        return null;
    }

}
```

Here is a trace of what the server returns for the quickfix:

```json
[Trace - 10:45:24 AM] Received response 'codeAction/resolve - (119)' in 5ms.
Result: {
    "title": "Create method 'bar()'",
    "kind": "quickfix",
    "diagnostics": [
        {
            "range": {
                "start": {
                    "line": 18,
                    "character": 8
                },
                "end": {
                    "line": 18,
                    "character": 11
                }
            },
            "severity": 1,
            "code": "67108964",
            "source": "Java",
            "message": "The method bar() is undefined for the type Function"
        }
    ],
    "edit": {
        "changes": {},
        "documentChanges": [
            {
                "textDocument": {
                    "version": null,
                    "uri": "file:///C:/Users/david/Documents/Programming/TestProjects/quarkus-func/src/main/java/functions/Function.java"
                },
                "edits": [
                    {
                        "snippet": {
                            "value": "${1:bar}();\n\n        return null;\n    }\n\n    ${2|private,public,protected|} ${3:void} ${1:bar}() {\n        // TODO Auto-generated method stub\n        throw new UnsupportedOperationException(\"Unimplemented method 'bar'\");\n    }"
                        },
                        "range": {
                            "start": {
                                "line": 18,
                                "character": 8
                            },
                            "end": {
                                "line": 21,
                                "character": 5
                            }
                        },
                        "newText": ""
                    }
                ]
            }
        ]
    },
    "command": {
        "title": "refresh Diagnostics",
        "command": "java.project.refreshDiagnostics",
        "arguments": [
            "file:///c%3A/Users/david/Documents/Programming/TestProjects/quarkus-func/src/main/java/functions/Function.java",
            "thisFile",
            false,
            true
        ]
    }
}
```

**Comment by rgrunber** [comment]
I'm seeing the exact same response, only file URI is different. When you have time, i would try to validate whether 

```diff
--- yours.txt	2024-09-26 11:17:54.751100419 -0400
+++ mine.txt	2024-09-26 11:23:02.727259008 -0400
@@ -1,4 +1,4 @@
-[Trace - 10:45:24 AM] Received response 'codeAction/resolve - (119)' in 5ms.
+[Trace - 11:22:36] Received response 'codeAction/resolve - (458)' in 39ms.
 Result: {
     "title": "Create method 'bar()'",
     "kind": "quickfix",
@@ -26,7 +26,7 @@
             {
                 "textDocument": {
                     "version": null,
-                    "uri": "file:///C:/Users/david/Documents/Programming/TestProjects/quarkus-func/src/main/java/functions/Function.java"
+                    "uri": "file:///home/rgrunber/sample-projects/my-quarkus-example/src/main/java/functions/Function.java"
                 },
                 "edits": [
                     {
@@ -53,7 +53,7 @@
         "title": "refresh Diagnostics",
         "command": "java.project.refreshDiagnostics",
         "arguments": [
-            "file:///c%3A/Users/david/Documents/Programming/TestProjects/quarkus-func/src/main/java/functions/Function.java",
+            "file:///home/rgrunber/sample-projects/my-quarkus-example/src/main/java/functions/Function.java",
             "thisFile",
             false,
             true

```

Definitely seems like something on the client side that's treated differently.

**Comment by testforstephen** [comment]
Looks like `snippet` text of "codeAction/resolve" response is not respected on Windows, and it still applies `newText`, which is `""` and behaves as delete.

**Comment by fp024** [comment]
Hello.

After upgrading the extension to version 1.35.0 in Windows 11 environment, Create local Variable '???' does not work properly.

However, there was no problem in Ubuntu 22.04 environment.

**Windows 11 + Ext Ver. 1.35.0**

https://github.com/user-attachments/assets/362dac89-a6e0-4e40-9c03-d2595178659c

**Ubuntu 22.04 + Ext Ver. 1.35.0** 

https://github.com/user-attachments/assets/f97e8079-5c60-4487-b43c-3d8184919673

**Comment by rgrunber** [comment]
@testforstephen , can you try the PR and let me know if it resolves it for you ? It should prevent `newText` from being populated.

If someone could try out https://github.com/rgrunber/vscode-java/releases/tag/v1.36.0 ( java-win32-x64-1.36.0-179.vsix) and let me know if that fixes things for them, that would help.

**Comment by fp024** [comment]
In my Windows 11 environment, installing 1.36.0 did not solve the problem.

![image](https://github.com/user-attachments/assets/c7c8c7b6-6e86-49b4-8fee-3464d780c526)

**Comment by rgrunber** [comment]
@testforstephen , I think we should just disable the feature ( `snippetEditSupport` in `extendedClientCapabilities` ) until there's some kind of fix upstream / we find a proper solution for windows. Could we do a bug-fix release for Monday ?

**Comment by testforstephen** [comment]
> @testforstephen , I think we should just disable the feature ( `snippetEditSupport` in `extendedClientCapabilities` ) until there's some kind of fix upstream / we find a proper solution for windows. Could we do a bug-fix release for Monday ?

+1

**Comment by rgrunber** [comment]
I found the root cause on Windows. It's our code :

https://github.com/redhat-developer/vscode-java/blob/0149fc7323f1a9202dab94fa10eda020d883b459/src/extension.ts#L334

URI from the language server is : `file:///C:/Users/user/projects/foo/Test.java`
URI after converting lsp.codeaction to a vscode.codeaction is : `file:///C%3A/Users/user/projects/foo/Test.java`

The conversion seems to be done by the `asCodeAction` conversion call that occurs higher up.

We can probably fix this by calling `Uri.parse(..)` on the initial Uri we retrieve from the language server.

Could someone try out https://github.com/rgrunber/vscode-java/releases/tag/v1.36.0 . It should contain a new vsix there that should work on Windows.

**Comment by datho7561** [comment]
@rgrunber It seems to be working:

https://github.com/user-attachments/assets/7fbd98d7-8cf2-4574-9fb7-f3347b97efa5

There's a bug related to the indentation of the inserted content. At the end of the video I formatted the document manually and the content seems to be correct after that

**Comment by rgrunber** [comment]
> There's a bug related to the indentation of the inserted content. At the end of the video I formatted the document manually and the content seems to be correct after that

This is https://github.com/microsoft/vscode/issues/63129

**Comment by fp024** [comment]
The v1.35.1 release yesterday worked well in a Windows environment. Thank you for your support. 👍

**Comment by iwangbowen** [comment]
Why it took so long to fix😡

**Comment by rgrunber** [comment]
1.35.0 was released on September 26th (Thursday). I think we were aware of the issue prior to release but didn't have enough information. The feature was available for about 20 days in the 1.35.0 pre-releases so had it been filed earlier, we probably could have fixed it well before the release.

We were investigating the problem the day of the release and by the next day, we decided it would be better to revert and see if we can re-enable for 1.36.0. We had a bug fix release (1.35.1) on Monday (30th).
